libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts

```diff
@@ -49,13 +47,14 @@ export class NewsletterComponent {
   private readonly _newsletterStore = inject(NewsletterStore);
   private readonly _router = inject(Router);
   private readonly _localizeRouter = inject(LocalizeRouterService);
-  private readonly _onSuccess = effect(() => {
-    if (this._newsletterStore.loading() === 'success') {
+  private readonly _onResult = effect(() => {
+    const loadingState = this._newsletterStore.loading();
+    if (loadingState === 'success' || loadingState === 'error') {
       this._router.navigate(
         this._localizeRouter.translateRoute(['/newsletter']) as string[],
         {
           queryParams: {
-            nm: 'confirmed',
+            nm: loadingState === 'success' ? 'confirmed' : 'not-confirmed',
```

Comment:
we navigate only for tracking purposes and don't wanna navigate user to that page on error. 
Error should be displayed down below the form.

Reply:
Corrected